pkgs/applications/networking/cluster/argo/default.nix

```diff
@@ -0,0 +1,24 @@
+{ lib, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
+  name = "argo-${version}";
+  version = "2.2.1";
+
+  src = fetchFromGitHub {
+    owner = "argoproj";
+    repo = "argo";
+    rev = "v${version}";
+    sha256 = "0x3aizwbqkg2712021wcq4chmwjhw2df702wbr6zd2a2cdypwb67";
+  };
+
+  goDeps = ./deps.nix;
+  goPackagePath = "github.com/argoproj/argo";
+
+  meta = {
```

Comment:
Most packages will do `with lib;` here so that they don’t have to repeat `lib` in this section.

Up to you, but thought I’d mention. :)

Reply:
Apologies, I meant like this:

    meta = with lib; {

Top-level imports can make scope confusing and should usually be avoided, but meta is so simple that it’s not an issue there.